### PR TITLE
Json component fixes

### PR DESCRIPTION
**Pull request by jerryedebua**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,54 @@
-# flow-json-component
\ No newline at end of file
+# Flow Json component
+The component is designed using Flow SDK to change values to json or reverse the process.
```
Comment (nawazdhandala) [inline README.md]: to convert text to JSON or to convert JSON to text,

File: README.md [hunk 2]
```diff
@@ -1 +1,54 @@
...
+```javascript
+// require the component
+const Component = require('flow-json-component');
+```
+
+*Create instance of component*
+
+```javascript
+// create instance of the toJson component to change content to json
+const component = new Component.toJson();
```
Comment (nawazdhandala) [inline README.md]: This documentation is wrong. You've made fundamental mistakes here. Rewrite.

File: components.json [hunk 3]
```diff
@@ -0,0 +1,142 @@
+{
+  "flow-json-component-to-json": {
+    "componentId": "flow-json-component-to-json",
+    "title": "To Json",
```
Comment (nawazdhandala) [inline components.json]: From text to JSON

File: components.json [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        "title": "Success",
+        "description": "Denotes the state when content is successfully changed to json",
+        "properties": [
+          {
+            "categoryName": "General",
+            "categoryDescription": "Basic settings about this port.",
+            "properties": [
+              {
+                "id": "Data",
+                "name": "Data",
```
Comment (nawazdhandala) [inline components.json]: JSON

File: components.json [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        "description": "Denotes the state when an error occurs",
+        "properties": [
+          {
+            "categoryName": "General",
+            "categoryDescription": "Basic settings about this port.",
+            "properties": [
+              {
+                "id": "Data",
+                "name": "Error",
+                "type": "object",
```
Comment (nawazdhandala) [inline components.json]: text -> throw a text if the string cant be converted - "Cannot convert to JSON"

File: components.json [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+      }
+    ],
+    "properties": [
+      {
+        "categoryName": "General",
+        "categoryDescription": "Basic settings about the ToJson component.",
+        "properties": [
+          {
+            "id": "Variable",
+            "name": "Content",
```
Comment (nawazdhandala) [inline components.json]: Text

File: components.json [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+            "type": "object",
```
Comment (nawazdhandala) [inline components.json]: text

File: components.json [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+            "data": null,
+            "options": { "required": true }
+          }
+        ]
+      }
+    ],
+    "showOnComponentsPanel": false,
```
Comment (nawazdhandala) [inline components.json]: true -> this should be true for all the components EXCEPT the start component. Fix this in all the component.json

File: components.json [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+    "isDeletable": false
```
Comment (nawazdhandala) [inline components.json]: true -> this should be true for all the components EXCEPT the start component. Fix this in all the component.json

File: components.json [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+    "file": "src/from-json.js",
+    "executionPlatform": ["server", "client"],
+    "metadata": {
+      "icon-color": "#7ab260"
+    },
+    "ports": [
+      {
+        "id": "Success",
+        "title": "Success",
+        "description": "Denotes the state when content is successfully parsed from json",
```
Comment (nawazdhandala) [inline components.json]: Converted Text from JSON

File: components.json [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+          }
+        ]
+      }
+    ],
+    "showOnComponentsPanel": false,
+    "isDeletable": false
+  },
+  "flow-json-component-from-json": {
+    "componentId": "flow-json-component-from-json",
+    "title": "From Json",
```
Comment (nawazdhandala) [inline components.json]: From JSON to Text

File: components.json [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+        "title": "Success",
+        "description": "Denotes the state when content is successfully parsed from json",
+        "properties": [
+          {
+            "categoryName": "General",
+            "categoryDescription": "Basic settings about this port.",
+            "properties": [
+              {
+                "id": "Data",
+                "name": "Data",
```
Comment (nawazdhandala) [inline components.json]: JSON

File: components.json [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+        "description": "Denotes the state when an error occurs",
+        "properties": [
+          {
+            "categoryName": "General",
+            "categoryDescription": "Basic settings about this port.",
+            "properties": [
+              {
+                "id": "Data",
+                "name": "Error",
+                "type": "object",
```
Comment (nawazdhandala) [inline components.json]: text

File: components.json [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+        ]
+      }
+    ],
+    "properties": [
+      {
+        "categoryName": "General",
+        "categoryDescription": "Basic settings about the FromJson component.",
+        "properties": [
+          {
+            "id": "Variable",
```
Comment (nawazdhandala) [inline components.json]: Text

File: components.json [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+            "name": "Content",
```
Comment (nawazdhandala) [inline components.json]: Text

File: components.json [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+            "type": "text",
+            "data": null,
+            "options": { "required": true }
+          }
+        ]
+      }
+    ],
+    "showOnComponentsPanel": false,
```
Comment (nawazdhandala) [inline components.json]: True for both of these.

File: src/from-json.js [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+    super();
+    this.name = 'From JSON';
+
+    this.attachTask(function() {
+      try {
+        const object = JSON.parse(this.getProperty('Variable').data);
+        const port = this.getPort('Success');
+        port.getProperty('Data').data = object;
+        port.emit();
+        this.taskComplete();
```
Comment (nawazdhandala) [inline src/from-json.js]: Have task complete outside of Try Catch

File: src/from-json.js [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+      } catch(err) {
+        const port = this.getPort('Error');
+        port.getProperty('Data').data = err;
+        port.emit();
+        this.taskComplete();
```
Comment (nawazdhandala) [inline src/from-json.js]: Have task complete outside of Try Catch

File: src/to-json.js [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+    super();
+    this.name = 'To JSON';
+
+    this.attachTask(function() {
+      try {
+        const json = JSON.stringify(this.getProperty('Variable').data);
+        const port = this.getPort('Success');
+        port.getProperty('Data').data = json;
+        port.emit();
+        this.taskComplete();
```
Comment (nawazdhandala) [inline src/to-json.js]: Have task complete outside of Try Catch

File: src/to-json.js [hunk 20]
```diff
@@ -0,0 +1,25 @@
...
+      } catch(err) {
+        const port = this.getPort('Error');
+        port.getProperty('Data').data = err;
+        port.emit();
+        this.taskComplete();
```
Comment (nawazdhandala) [inline src/to-json.js]: Have task complete outside of Try Catch

File: components.json (hunk 12 above)
Comment (jerryedebua) [reply components.json]: Don't you mean Text here for this port? Cause we are converting from JSON to Text

**Pull request merged**